src/Webapi/Temporal/Webapi__Temporal__Instant.res

```diff
@@ -0,0 +1,92 @@
+type t = Webapi__Temporal__Types.instant
+
+// @new external make: BigInt.t => t = "Temporal.Instant"
+@scope("Temporal.Instant") @val external fromString: string => t = "from"
+@scope("Temporal.Instant") @val external fromInstant: t => t = "from"
+@scope("Temporal.Instant") @val external fromEpochSeconds: float => t = "fromEpochSeconds"
+@scope("Temporal.Instant") @val
+external fromEpochMilliseconds: float => t = "fromEpochMilliseconds"
+// @scope("Temporal.Instant") @val external fromEpochMicroseconds: BigInt.t => t = "fromEpochMicroseconds"
+// @scope("Temporal.Instant") @val external fromEpochNanoseconds: BigInt.t => t = "fromEpochNanoseconds"
+@scope("Temporal.Instant") @val external compare: (t, t) => int = "compare"
+@scope("Temporal.Instant") @val external compareStrings: (string, string) => int = "compare"
+
+@get external epochSeconds: t => float = "epochSeconds"
+@get external epochMilliseconds: t => float = "epochMilliseconds"
+// @get external epochMicroseconds: t => BigInt.t = "epochMicroseconds"
+// @get external epochNanoseconds: t => BigInt.t = "epochNanoseconds"
+@send
+external toZonedDateTimeISO: (
+  t,
+  Webapi__Temporal__Types.timeZone,
+) => Webapi__Temporal__ZonedDateTime.t = "toZonedDateTimeISO"
+@send
+external toZonedDateTime: (
+  t,
+  {"timeZone": Webapi__Temporal__Types.timeZone, "calendar": Webapi__Temporal__Types.calendar},
+) => Webapi__Temporal__ZonedDateTime.t = "toZonedDateTime"
+@send external add: (t, Webapi__Temporal__Types.duration) => t = "add"
+@send external subtract: (t, Webapi__Temporal__Types.duration) => t = "subtract"
+
+@send external until: (t, t) => Webapi__Temporal__Types.duration = "until"
+@send
+external untilWithOptions: (
+  t,
+  t,
+  Webapi__Temporal__Difference.DifferenceOptions.t,
+) => Webapi__Temporal__Types.duration = "until"
+@send external since: (t, t) => Webapi__Temporal__Types.duration = "since"
+@send
+external sinceWithOptions: (
+  t,
+  t,
+  Webapi__Temporal__Difference.DifferenceOptions.t,
+) => Webapi__Temporal__Types.duration = "since"
+
+type roundTo = [
+  | #hour
+  | #minute
+  | #second
+  | #millisecond
+  | #microsecond
+  | #nanosecond
+]
+
+module RoundOptions = {
+  type t
+
+  @obj
+  external make: (
+    ~smallestUnit: roundTo,
+    ~roundingIncrement: int=?,
+    ~roundingMode: [#halfExpand | #ceil | #trunc | #floor]=?,
+    unit
+  ) => t = ""
+}
+
+@send external round: (t, ~to: roundTo) => t = "round"
+@send external roundWithOptions: (t, RoundOptions.t) => t = "round"
```

Comment:
If you're careful the sub-module can be avoided for simple cases like this. I know the rest of the library does it this way but once we look at modernising the API (and reducing runtime code generation) I might try to rename some of the `make` submodule methods to be named directly, instead of encapsulated:

```suggestion
type roundOptions

@obj
external roundOptions: (
  ~smallestUnit: roundTo,
  ~roundingIncrement: int=?,
  ~roundingMode: [#halfExpand | #ceil | #trunc | #floor]=?,
  unit
) => roundOptions = ""

@send external round: (t, ~to: roundTo) => t = "round"
@send external roundWithOptions: (t, roundOptions) => t = "round"
```

And so on for the other submodules you've added in this PR. In this style using it becomes `value->Instant.roundWithOptions(Instant.roundOptions(~smallestUnit=#hour, ()))`.

Although to be fair by the time we get to v2 we might have ReScript 10 available, and if the structural typing change in master works as well as we hope the entire `@obj` system will be replaced.